File: back/app/models/Exame.py
# --- Atributos ---
# Exames: Código do Exame, Descrição, Código Especialidade, Valor do Exame

class Exame:
    """Representa a entidade Exame no sistema."""

    def __init__(self, codigo_exame: int, descricao: str, codigo_especialidade: int, valor_exame: float):
        self.codigo_exame = codigo_exame
        self.descricao = descricao
        self.codigo_especialidade = codigo_especialidade  # Chave estrangeira para Especialidade
        self.valor_exame = valor_exame  # Valor cobrado pelo exame

    def __str__(self):
        """Retorna uma representação legível do objeto Exame."""
        return (f"Cód Exame: {self.codigo_exame}, Descrição: {self.descricao}, "
                f"Especialidade Cód: {self.codigo_especialidade}, Valor: R$ {self.valor_exame:.2f}")

    def to_dict(self):
        """Retorna um dicionário com os atributos do exame."""
        return {
            "codigo_exame": self.codigo_exame,
            "descricao": self.descricao,
            "codigo_especialidade": self.codigo_especialidade,
            "valor_exame": self.valor_exame
        }

File: front/src/app/core/model/exame/Exame.ts
export default class Exame {
  private _codigo_exame: number;
  private _descricao: string;
  private _valor_exame: number;
  private _codigo_especialidade: number;


  constructor(
    codigo_exame: number = 0,
    descricao: string = '',
    valor_exame: number = 0,
    codigo_especialidade: number = 0
  ) {
    this._codigo_exame = codigo_exame;
    this._descricao = descricao;
    this._valor_exame = valor_exame;
    this._codigo_especialidade = codigo_especialidade;
  }


  get codigo_exame(): number {
    return this._codigo_exame;
  }

  set codigo_exame(value: number) {
    this._codigo_exame = value;
  }

  get descricao(): string {
    return this._descricao;
  }

  set descricao(value: string) {
    this._descricao = value;
  }

  get valor_exame(): number {
    return this._valor_exame;
  }

  set valor_exame(value: number) {
    this._valor_exame = value;
  }

  get codigo_especialidade(): number {
    return this._codigo_especialidade;
  }

  set codigo_especialidade(value: number) {
    this._codigo_especialidade = value;
  }

  toJSON() {
    return {
      codigo_exame: String(this._codigo_exame),
      descricao: this._descricao,
      valor_exame: this._valor_exame,
      codigo_especialidade: this._codigo_especialidade
    };
  }
}
